src/api/participantsRouter.ts

```diff
@@ -30,169 +31,185 @@ import {
   getAllUserFromParticipant,
 } from './services/usersService';
 
-export const participantsRouter = express.Router();
-
-participantsRouter.get('/available', async (_req, res) => {
-  const participants = await Participant.query().whereNotNull('siteId').withGraphFetched('types');
-  return res.status(200).json(participants);
-});
-
-participantsRouter.post('/', async (req, res) => {
-  try {
-    const data = {
-      ...ParticipantCreationPartial.parse(req.body),
-      status: ParticipantStatus.AwaitingApproval,
-    };
-    // insertGraphAndFetch will implicitly create a transaction
-    const newParticipant = await Participant.query().insertGraphAndFetch([data], {
-      relate: true,
-    });
-
-    sendNewParticipantEmail(
-      data,
-      data.types.map((t) => t.id)
-    );
-    return res.status(201).json(newParticipant);
-  } catch (err) {
-    if (err instanceof z.ZodError) {
-      return res.status(400).send(err.issues);
+export type AvailableParticipant = Pick<ParticipantDTO, 'id' | 'name' | 'siteId' | 'types'>;
+function mapParticipantToAvailableParticipant(participant: Participant) {
+  return {
+    id: participant.id,
+    name: participant.name,
+    siteId: participant.siteId,
+    types: participant.types,
+  };
+}
+
+export function createParticipantsRouter() {
+  const participantsRouter = express.Router();
+
+  participantsRouter.get('/available', async (_req, res) => {
+    const participants = await Participant.query().whereNotNull('siteId').withGraphFetched('types');
+    const participantDTOs = participants.map(mapParticipantToAvailableParticipant);
```

Comment:
querybuilder can chain .castTo<someType>()  to change the type of returning
like: `await Participant.query().whereNotNull('siteId').withGraphFetched('types').castTo<AvailableParticipant>()`